scripts/BuildLibsRelease.sh

```diff
@@ -1,113 +1,157 @@
-sudo mkdir -p /opt/local/lib
-sudo mkdir -p /opt/local/bin
+#!/bin/bash -e
 
-#setup permissions on /opt/local
-sudo chgrp -R staff /opt/local*
-sudo chmod -R g+w /opt/local*
+# System dependency checks...
+if [ ! -d /opt/local/lib ] || [ ! -d /opt/local/libdbg ] || [ ! -d /opt/local/bin ] || [ ! -d /opt/local/bin ] ; then
+  echo "/opt/local/bin, /opt/local/lib, /opt/local/libdbg and /opt/local/include must exist and be writable!"
+  exit 1
+fi
+if ! command -v brew > /dev/null ; then
+  echo 'Some libraries require Homebrew Tools, to install Homebrew:'
+  echo '/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"'
+  exit 1
+fi
 
-export MACOSX_DEPLOYMENT_TARGET=10.14
+export MACOSX_DEPLOYMENT_TARGET=${MACOSX_DEPLOYMENT_TARGET:-11.0}
 export OSX_VERSION_MIN="-mmacosx-version-min=${MACOSX_DEPLOYMENT_TARGET}"
-export ARM64_TARGETS="-target arm64-apple-macos10.14 -arch arm64"
-export X86_64_TARGETS="-target x86_64-apple-macos10.14 -arch x86_64"
+export ARM64_TARGETS="-target arm64-apple-macos11.0 -arch arm64"
+export X86_64_TARGETS="-target x86_64-apple-macos11.0 -arch x86_64"
+
+# Absolute path to directory containing the xLights source tree
+# Default assumes that xlights is cloned in your cwd
+export XL_DIR="${XL_DIR:-$(realpath ./xlights)}"
+
+# Core count of the build host
+export NUM_CPUS="${NUM_CPUS:-$(sysctl -n hw.ncpu)}"
+echo "Building with $NUM_CPUS cores"
 
 #need ONE of these lines
 export XL_TARGETS="${X86_64_TARGETS} ${ARM64_TARGETS}"
 # export XL_TARGETS="${X86_64_TARGETS}"
 # export XL_TARGETS="${ARM64_TARGETS}"
 
-# need ONE of these
-# export BUILD_HOST=x86_64
-export BUILD_HOST=arm
+export BUILD_HOST="$(uname -p)"
+echo "Build host architecture is $BUILD_HOST"
 
-#Note: Some libraries require Homebrew Tools, to install Homebrew:
-/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
-
-#Autotools and LibTool are required from Homebrew:
+set -x
+# install dependencies from homebrew
 brew install automake libtool cmake nasm
 
 echo "libzstd"
-git clone https://github.com/facebook/zstd
-cd zstd
-git checkout v1.5.2
+if [ ! -d "zstd" ]; then git clone https://github.com/facebook/zstd ; fi
+pushd zstd
+git checkout v1.5.6
 export CFLAGS="-g -flto=thin  ${OSX_VERSION_MIN} ${XL_TARGETS}"
 export LDFLAGS="-flto=thin  ${OSX_VERSION_MIN} ${XL_TARGETS} "
 make clean
-make -j 8 HAVE_LZMA=0 HAVE_LZ4=0 lib-mt
+make -j ${NUM_CPUS} HAVE_LZMA=0 HAVE_LZ4=0 lib-mt
 cp lib/libzstd.a /opt/local/lib
+export CFLAGS="-g  ${OSX_VERSION_MIN} ${XL_TARGETS}"
+export LDFLAGS=" ${OSX_VERSION_MIN} ${XL_TARGETS}"
+make clean
+make -j ${NUM_CPUS} HAVE_LZMA=0 HAVE_LZ4=0 lib-mt
+cp lib/libzstd.a /opt/local/libdbg
 unset CFLAGS
```

Comment:
might be worth shelling out her instead of trying to figure out how to revert directory switching+env var changes+etc

Reply:
This basically duplicates:
https://github.com/xLightsSequencer/xLights-macOS/pull/5/files#r1867865797